ProjFS.Mac/PrjFSKext/Message_Kernel.cpp

```diff
@@ -33,7 +35,57 @@ void Message_Init(
     {
         header->pathSizeBytes = 0;
     }
-    
+
+    if (nullptr != fromPath)
+    {
+        header->fromPathSizeBytes = strlen(fromPath) + 1;
+    }
+    else
+    {
+        header->fromPathSizeBytes = 0;
+    }
+
     spec->messageHeader = header;
     spec->path = path;
+    spec->fromPath = fromPath;
+}
+
+uint32_t Message_EncodedSize(const Message& message)
+{
+    return sizeof(*message.messageHeader) + message.messageHeader->pathSizeBytes + message.messageHeader->fromPathSizeBytes;
+}
+
+uint32_t Message_Encode(void* buffer, const uint32_t bufferSize, const Message& message)
+{
+    uint8_t* bufferPosition = static_cast<uint8_t*>(buffer);
+    uint32_t bufferBytesRemain = bufferSize;
+    
+    assert(bufferSize >= sizeof(*message.messageHeader));
+    memcpy(bufferPosition, message.messageHeader, sizeof(*message.messageHeader));
+    bufferPosition +=    sizeof(*message.messageHeader);
```

Comment:
nit: spacing

Reply:
I guess I find it helpful to make it extra clear that we're incrementing the position and decrementing the remaining byte count by the same amount by aligning the amounts.

Reply:
i've never held up a PR over spacing, so I wouldn't do that here.
To me it looks more like off spacing than pointing out the +/-.
Just 1 person's opinion :-)